src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/stores/ResourceFormStore.js

```diff
@@ -87,15 +87,139 @@ export default class ResourceFormStore extends AbstractFormStore implements Form
         });
     };
 
-    @action handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
+    handleSchemaResponse = ([schema, jsonSchema]: [Schema, Object]) => {
         this.validator = jsonSchema ? ajv.compile(jsonSchema) : undefined;
         this.pathsByTag = {};
 
-        this.schema = schema;
-        this.addMissingSchemaProperties();
-        this.setSchemaLoading(false);
+        return this.loadAndMergeRemoteData(this.schema, schema).then(action(() => {
+            this.schema = schema;
+            this.addMissingSchemaProperties();
+            this.validate();
+            this.setSchemaLoading(false);
+        }));
     };
 
+    loadAndMergeRemoteData = (localSchema: Schema, remoteSchema: Schema) => {
+        // load data only after initial schema was set to prevent duplicate requests during initialization
+        if (localSchema) {
+            return this.resourceStore.requestRemoteData({template: this.type}).then((data: Object) => {
+                const result = this.mergeData(localSchema, remoteSchema, this.data, data);
+                this.setMultiple(result);
+            });
+        }
+        return Promise.resolve();
+    };
+
+    mergeData(
+        localSchema: Schema,
+        remoteSchema: Schema,
+        localData: Object,
+        remoteData: Object
+    ) {
+        let result = {};
+        if (!localSchema || !remoteSchema) {
+            return result;
+        }
+
+        for (const name in remoteSchema) {
+            const {
+                items: remoteItems,
+                defaultType: remoteDefaultType,
+                type: remoteType,
+                types: remoteTypes,
+            } = remoteSchema[name];
+            const {
+                items: localItems,
+                defaultType: localDefaultType,
+                type: localType,
+                types: localTypes,
+            } = localSchema[name] || {};
+
+            if (remoteType === SECTION_TYPE && remoteItems &&
```

Comment:
think this is not needed because the conditions below will handle this case implicitly

Reply:
I don't think so, because in the conditions below we pass either `localSchema`/`remoteItems` or `localItems`/`remoteItems` but never `localItems/remoteItems` and only one condition is called because of the `continue`